subsys/usb/device_next/class/usbd_uvc.c

```diff
@@ -0,0 +1,2335 @@
+	uint16_t bmRateControlModes;
+	uint64_t bmLayoutPerStream;
+} __packed;
+
+struct uvc_payload_header {
+	uint8_t bHeaderLength;
+	uint8_t bmHeaderInfo;
+	uint32_t dwPresentationTime; /* optional */
+	uint32_t scrSourceClockSTC;  /* optional */
+	uint16_t scrSourceClockSOF;  /* optional */
+} __packed;
+
+/* Information specific to each VideoStreaming interface */
+struct uvc_stream {
+	const struct device *dev;
+	atomic_t state;
+	struct k_work work;
+	struct k_fifo fifo_in;
+	struct k_fifo fifo_out;
+	struct usbd_uvc_strm_desc *desc;
+	struct uvc_probe default_probe;
+	struct uvc_payload_header payload_header;
+	uint8_t format_id;
+	uint8_t frame_id;
+	const struct device *video_dev;
+	struct video_format video_fmt;
+	struct video_frmival video_frmival;
+	size_t vbuf_offset;
+};
+
+/* Global information */
+struct uvc_data {
+	atomic_t state;
+	struct usbd_class_data *c_data;
+	struct uvc_stream *streams;
+	struct usbd_uvc_desc *desc;
+	struct usb_desc_header **fs_desc;
+	struct usb_desc_header **hs_desc;
+	/* UVC error from latest request */
+	int err;
+};
+
+/* Specialized version of UDC net_buf metadata with extra fields */
+struct uvc_buf_info {
+	/* Regular UDC buf info so that it can be passed to USBD directly */
+	struct udc_buf_info udc;
+	/* Extra field at the end */
+	struct video_buffer *vbuf;
+	/* UVC stream this buffer belongs to */
+	struct uvc_stream *stream;
+} __packed;
+
+/* Mapping between UVC controls and Video controls */
+struct uvc_control_map {
+	uint8_t selector;
+	uint8_t size;
+	uint8_t bit;
+	enum uvc_op op;
+	uint32_t param;
+};
+
+struct uvc_guid_quirk {
+	uint32_t fourcc;
+	uint8_t guid[16];
+};
+
+typedef int uvc_control_fn_t(struct uvc_stream *strm, uint8_t selector, uint8_t request,
+			     struct net_buf *buf);
+
+NET_BUF_POOL_VAR_DEFINE(uvc_pool, DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) * 16,
+			512 * DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) * 16,
+			sizeof(struct uvc_buf_info), NULL);
+
+/* UVC helper functions */
+
+static const struct uvc_guid_quirk uvc_guid_quirks[] = {
+	{VIDEO_PIX_FMT_YUYV,  UVC_FORMAT_GUID("YUY2")},
+};
+
+static void uvc_fourcc_to_guid(uint8_t guid[16], uint32_t fourcc)
+{
+	for (int i = 0; i < ARRAY_SIZE(uvc_guid_quirks); i++) {
+		if (uvc_guid_quirks[i].fourcc == fourcc) {
+			memcpy(guid, uvc_guid_quirks[i].guid, 16);
+			return;
+		}
+	}
+	fourcc = sys_cpu_to_le32(fourcc);
+	memcpy(guid, UVC_FORMAT_GUID("...."), 16);
+	memcpy(guid, &fourcc, 4);
+}
+
+static uint32_t uvc_guid_to_fourcc(uint8_t guid[16])
+{
+	uint32_t fourcc;
+
+	for (int i = 0; i < ARRAY_SIZE(uvc_guid_quirks); i++) {
+		if (memcmp(guid, uvc_guid_quirks[i].guid, 16) == 0) {
+			return uvc_guid_quirks[i].fourcc;
+		}
+	}
+
+	memcpy(&fourcc, guid, 4);
+
+	return sys_le32_to_cpu(fourcc);
+}
+
+/* UVC control handling */
+
+static const struct uvc_control_map uvc_control_map_ct[] = {
+	{.size = 1, .bit = 1, .selector = UVC_CT_AE_MODE_CONTROL,
+	 .op = UVC_OP_VC_FIXED, .param = BIT(0)},
```

Comment:
There are only a few places with these squeezed initialization. This is sometimes okay, but it does not follow the usual style and is inconsistent, please rework them.
```c
static const struct uvc_control_map uvc_control_map_ct[] = {
	{
		.size = 1,
		.bit = 1,
		.selector = UVC_CT_AE_MODE_CONTROL,
	 	.op = UVC_OP_VC_FIXED,
		.param = BIT(0)
	},
```